modules/terraform/options.go

```diff
@@ -102,3 +106,144 @@ func WithDefaultRetryableErrors(t *testing.T, originalOptions *Options) *Options
+	}
+
+	return fmt.Sprintf("%v", variable), nil
+}
+
+// GetVariableAsMapFromVarFile Gets the map represention of a variable from a provided input file found in VarFile
+func GetVariableAsMapFromVarFile(t *testing.T, option *Options, fileName string, key string) map[string]string {
+	result, err := GetVariableAsMapFromVarFileE(t, option, fileName, key)
+	require.NoError(t, err)
+
+	return result
+}
+
+// GetVariableAsMapFromVarFileE Gets the map represention of a variable from a provided input file found in VarFile
+// Returns an error if GetAllVariablesFromVarFileE returns an error, the key provided does not exist, or the value associated with the key is not a map
+func GetVariableAsMapFromVarFileE(t *testing.T, option *Options, fileName string, key string) (map[string]string, error) {
+	resultMap := make(map[string]string)
+	variables, err := GetAllVariablesFromVarFileE(t, option, fileName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	_, exists := variables[key]
+
+	if !exists {
+		return nil, InputFileKeyNotFound(key)
+	}
+
+	if reflect.TypeOf(variables[key]).String() != "[]map[string]interface {}" {
+		return nil, UnexpectedOutputType{Key: key, ExpectedType: "map", ActualType: reflect.TypeOf(variables[key]).String()}
+	}
+
+	mapKeys := variables[key].([]map[string]interface{})[0]
+
+	for mapKey, mapVal := range mapKeys {
+		resultMap[mapKey] = fmt.Sprintf("%v", mapVal)
+	}
+
+	return resultMap, nil
+}
+
+// GetVariableAsListFromVarFile Gets the string list represention of a variable from a provided input file found in VarFile
+func GetVariableAsListFromVarFile(t *testing.T, option *Options, fileName string, key string) []string {
+	result, err := GetVariableAsListFromVarFileE(t, option, fileName, key)
+	require.NoError(t, err)
+
+	return result
+}
+
+// GetVariableAsListFromVarFileE Gets the string list represention of a variable from a provided input file found in VarFile
+// Will return error if GetAllVariablesFromVarFileE returns an error, the key provided does not exist, or the value associated with the key is not a list
+func GetVariableAsListFromVarFileE(t *testing.T, option *Options, fileName string, key string) ([]string, error) {
+	resultArray := []string{}
+	variables, err := GetAllVariablesFromVarFileE(t, option, fileName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if _, exists := variables[key]; !exists {
+		return nil, InputFileKeyNotFound(key)
+	}
+
+	if reflect.TypeOf(variables[key]).String() != "[]interface {}" {
+		return nil, UnexpectedOutputType{Key: key, ExpectedType: "list", ActualType: reflect.TypeOf(variables[key]).String()}
+	}
+
+	for _, item := range variables[key].([]interface{}) {
+		resultArray = append(resultArray, fmt.Sprintf("%v", item))
+	}
+
+	return resultArray, nil
+}
+
+// GetAllVariablesFromVarFile Parses all data from a provided input file found in VarFile and returns them as a key-value map
+func GetAllVariablesFromVarFile(t *testing.T, option *Options, fileName string) map[string]interface{} {
```

Comment:
A `map[string]interface{}` is a bit unwiedly and error prone to use, as you have to cast all the values within the map. What are your thoughts instead on an interface like:

```
func GetAllVariablesFromVarFile(t *testing.T, option *Options, fileName string, returnValue interface{})
```

Where `returnValue` is a pointer the user can pass in with any custom struct, and we will parse the tfvar file into that struct, including all the types within it. This is similar to `OutputStruct`.

Reply:
That's valid. It would certainly make it easier for the user. I did see to minimize using pointers in the best practices for this repo and this is a bit unique compared to the other functions, but it probably makes sense given the complexity of the return type.